src/main/java/com/google/devtools/build/lib/buildtool/ExecutionTool.java

```diff
@@ -725,6 +725,7 @@ private ImmutableList<ConvenienceSymlink> createConvenienceSymlinks(
     Set<BuildConfigurationValue> targetConfigurations =
         buildRequestOptions.useTopLevelTargetsForSymlinks()
             ? analysisResult.getTargetsToBuild().stream()
+                .map(ConfiguredTarget::getActual)
```

Comment:
What does `bazel build //:all` report in this context? Does it automatically fold the alias and actual into the same output report or does it keep them as separate entries?

Reply:
It looks like with and without my change the behavior when building both the alias and the target (from my example in the issue) is the same:

```
% bazel-dev build :all --config=ora --use_top_level_targets_for_symlinks
INFO: Analyzed 2 targets (1 packages loaded, 4 targets configured).
INFO: Found 2 targets...
INFO: Elapsed time: 0.111s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
% bazelisk build :all --config=ora --use_top_level_targets_for_symlinks
Starting local Bazel server and connecting to it...
INFO: Analyzed 2 targets (59 packages loaded, 976 targets configured).
INFO: Found 2 targets...
WARNING: cleared convenience symlink(s) bazel-bin, bazel-testlogs because their destinations would be ambiguous
INFO: Elapsed time: 7.108s, Critical Path: 0.38s
INFO: 8 processes: 4 internal, 3 darwin-sandbox, 1 local.
INFO: Build completed successfully, 8 total actions
```

Reply:
before:

```
% bazelisk build :cli :alias  --use_top_level_targets_for_symlinks --show_result=2
Starting local Bazel server and connecting to it...
INFO: Analyzed 2 targets (59 packages loaded, 976 targets configured).
INFO: Found 2 targets...
WARNING: cleared convenience symlink(s) bazel-bin, bazel-testlogs because their destinations would be ambiguous
Target //:cli up-to-date:
  bazel-out/applebin_macos-darwin_arm64-fastbuild-ST-9e7e65650639/bin/cli
Target //:cli up-to-date:
  bazel-out/applebin_macos-darwin_arm64-fastbuild-ST-9e7e65650639/bin/cli
INFO: Elapsed time: 7.179s, Critical Path: 0.39s
INFO: 8 processes: 4 internal, 3 darwin-sandbox, 1 local.
INFO: Build completed successfully, 8 total actions
```

after:

```
% bazel-dev build :cli :alias  --use_top_level_targets_for_symlinks --show_result=2
INFO: Analyzed 2 targets (0 packages loaded, 0 targets configured).
INFO: Found 2 targets...
Target //:cli up-to-date:
  bazel-bin/cli
Target //:cli up-to-date:
  bazel-bin/cli
INFO: Elapsed time: 0.083s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
```

Reply:
(all vs specifying multiple targets is the same)

Reply:
So this is an improvement for this, as previously the alias vs target were different configurations it seems

Reply:
Interesting. The reporting is still a big confusing since it doesn't acknowledge why `//:cli` is reported twice. But that's orthogonal to what you're doing - just an observation.

Thanks for the explanation. I don't see any scenario where the alias's direct configuration is relevant for this.